### PR TITLE
Docs: Explain the usage of logical expressions for `create_aggregate_expr`

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

part of #11359

### CODE REVIEW

**Review by dharanad — APPROVED** [review]
LGTM

File: datafusion/physical-expr-common/src/aggregate/mod.rs [hunk 1]
```diff
@@ -43,6 +43,9 @@ use datafusion_expr::utils::AggregateOrderSensitivity;
 
 /// Creates a physical expression of the UDAF, that includes all necessary type coercion.
 /// This function errors when `args`' can't be coerced to a valid argument type of the UDAF.
+///
+/// `input_exprs` and `sort_exprs` are used for customizing Accumulator as the arguments in `AccumulatorArgs`,
```
Comment (alamb) [inline datafusion/physical-expr-common/src/aggregate/mod.rs:47]:
```suggestion
/// `input_exprs` and `sort_exprs` are used for customizing Accumulatorsz
/// whose behavior depends on arguments such as a an `ORDER BY`.
///
/// For example to call `ARRAY_AGG(x ORDER BY y)` would pass `[y]` 
/// `sort_exprs`
```

**Review by alamb — APPROVED** [review]
Thank you @jayzhan211  -- I left a suggestion for an example -- let me know if I got it wrong

**Comment by jayzhan211** [comment]
Actually if others call this function to create physical expression directly, they can just provide empty slice. The logical expressions are used internally only.

**Review by alamb — APPROVED** [review]

**Pull request merged**